### PR TITLE
Added Guest button to login screen

### PR DESCRIPTION
**Pull request by rkuenzi-epfl**
Also modified login screen tests so that it disconnect potential signed
in google user before the test.

### CODE REVIEW

File: app/src/androidTest/java/com/github/wnder/LoginActivityTest.java [hunk 1]
```diff
@@ -1,56 +1,94 @@
...
+        GoogleSignInClient client = GoogleSignIn.getClient(ctx, gso);
+        Task<Void> taskWaiter;
+        if(GoogleSignIn.getLastSignedInAccount(ctx) != null){
+            taskWaiter = client.signOut();
+            try {
+                // Wait maximum of 30 seconds, otherwise assume the user was disconnected successfully
+                await(taskWaiter, 30, TimeUnit.SECONDS);
+            } catch (Exception e){
+                System.out.println("Assuming disconnection succeeded");
+            }
```
Comment (leonardvaney) [inline app/src/androidTest/java/com/github/wnder/LoginActivityTest.java:71]: Probably not the best way to do it but if the test passes for most of the time it's fine.

**Review by leonardvaney — APPROVED** [review]

File: app/src/main/res/layout/activity_login.xml [hunk 2]
```diff
@@ -27,4 +27,15 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/textLogin" />
 
+    <Button
+        android:id="@+id/guest_button"
```
Comment (JeremyDezalos) [inline app/src/main/res/layout/activity_login.xml:31]: Didn't we have a convention to use camel case on ids ?

File: app/src/main/res/layout/activity_login.xml [hunk 3]
```diff
@@ -27,4 +27,15 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="32dp"
+        android:text="Guest"
+        app:layout_constraintEnd_toEndOf="@id/sign_in_button"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toStartOf="@id/sign_in_button"
+        app:layout_constraintTop_toBottomOf="@+id/sign_in_button" />
```
Comment (JeremyDezalos) [inline app/src/main/res/layout/activity_login.xml:39]: Same than before

File: app/src/main/java/com/github/wnder/LoginActivity.java [hunk 4]
```diff
@@ -33,6 +33,8 @@ protected void onCreate(Bundle savedInstanceState) {
         signInButton = findViewById(R.id.sign_in_button);
         signInButton.setVisibility(View.INVISIBLE);  // Hide the button
         signInButton.setOnClickListener(this);
+        findViewById(R.id.guest_button).setOnClickListener(this);
```
Comment (JeremyDezalos) [inline app/src/main/java/com/github/wnder/LoginActivity.java:36]: Shouldn't we provide onClickListeners instead of an Activity ?

**Review by JeremyDezalos — APPROVED** [review]

File: app/src/main/java/com/github/wnder/LoginActivity.java (hunk 4 above)
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/LoginActivity.java:36]: I admit this looks a bit strange but it allows to use a single onClick method in the activity to handle any button click with a simple switch statement so I think it has its advantages.

**Pull request merged**